pkg/networkpolicy/adminnetworkpolicy.go

```diff
@@ -0,0 +1,282 @@
+package networkpolicy
+
+import (
+	"cmp"
+	"fmt"
+	"net"
+	"slices"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+	npav1alpha1 "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+)
+
+// adminNetworkPolicyAction evaluate the Admin Network Policies on the packet
+// that is evaluated both from the origin and the destination perspective.
+// There can be 3 possible outcomes:
+// 1. Policies Allow in both direction then the packet is Allowed and NOT evaluated by Network Policies
+// 2. Policies Deny in any direction then the packet is Dropped and NOT evaluated by Network Policies
+// 3. Policies Pass and or Allow then the packet is passed to be evaluated by Network Policies
+func (c *Controller) adminNetworkPolicyAction(p packet) npav1alpha1.AdminNetworkPolicyRuleAction {
+	srcIP := p.srcIP
+	srcPod := c.getPodAssignedToIP(srcIP.String())
+	srcPort := p.srcPort
+	dstIP := p.dstIP
+	dstPod := c.getPodAssignedToIP(dstIP.String())
+	dstPort := p.dstPort
+	protocol := p.proto
+	srcPodAdminNetworkPolices := c.getAdminNetworkPoliciesForPod(srcPod)
+	dstPodAdminNetworkPolices := c.getAdminNetworkPoliciesForPod(dstPod)
+
+	msg := fmt.Sprintf("checking packet %s", p.String())
+	if srcPod != nil {
+		msg += fmt.Sprintf("\nSrcPod (%s/%s): %d AdminNetworkPolicy", srcPod.Name, srcPod.Namespace, len(srcPodAdminNetworkPolices))
+	}
+	if dstPod != nil {
+		msg += fmt.Sprintf("\nDstPod (%s/%s): %d AdminNetworkPolicy", dstPod.Name, dstPod.Namespace, len(dstPodAdminNetworkPolices))
+	}
+	klog.V(2).Infof("%s", msg)
+
+	egressAction := c.evaluateAdminEgress(srcPodAdminNetworkPolices, dstPod, dstIP, dstPort, protocol)
+	ingressAction := c.evaluateAdminIngress(dstPodAdminNetworkPolices, srcPod, srcPort, protocol)
+
+	// Traffic is denied in at least one direction
+	if ingressAction == npav1alpha1.AdminNetworkPolicyRuleActionDeny ||
+		egressAction == npav1alpha1.AdminNetworkPolicyRuleActionDeny {
+		return npav1alpha1.AdminNetworkPolicyRuleActionDeny
+	}
+
+	// Traffic is allowed in both direction
+	if ingressAction == npav1alpha1.AdminNetworkPolicyRuleActionAllow &&
+		egressAction == npav1alpha1.AdminNetworkPolicyRuleActionAllow {
+		return npav1alpha1.AdminNetworkPolicyRuleActionAllow
+	}
+
+	// Traffic has to pass to be passed to Network Policies
+	return npav1alpha1.AdminNetworkPolicyRuleActionPass
```

Comment:
@npinaeva @tssurya see the logic here, 
1. we evaluate if the srcPod has any policy in egress to the dstPod and get its action
2. we evaluate if the dstPod has any policy in ingress from the srcPod and get its action
3. If there is at least one deny we just drop
4. if both are allow then we allow
5. the rest means it has to passed down to keep evaluating by network policies

Reply:
I think we are still missing some allow conditions, mainly when one direction ANP says Pass and another says Allow. To check that we need a test where both ANPs and NPs are present, example from the previous comment should still work.